bootstrap/httpserver/server_test.go

```diff
@@ -0,0 +1,207 @@
+	logger := log.GetLogger().WithOutput(log.NewOutput(os.Stdout).WithFormatter(log.NewHumanReadableFormatter()))
+	papiMock := &services.MockPublicApi{}
+
+	s := NewHttpServer("", logger, papiMock)
+
+	m := client.CallMethodRequestReader([]byte("Random Bytes"))
+
+	rec := httptest.NewRecorder()
+	s.(*server).isValid(m, rec)
+
+	require.Equal(t, http.StatusBadRequest, rec.Code, "bad input in body should cause bad request error")
+}
+
+func Test_HttpServer_TranslateStatusToHttpCode(t *testing.T) {
+	tests := []struct {
+		name   string
+		expect int
+		status protocol.RequestStatus
+	}{
+		{"REQUEST_STATUS_RESERVED", http.StatusInternalServerError, protocol.REQUEST_STATUS_RESERVED},
+		{"REQUEST_STATUS_COMPLETED", http.StatusOK, protocol.REQUEST_STATUS_COMPLETED},
+		{"REQUEST_STATUS_IN_PROCESS", http.StatusAccepted, protocol.REQUEST_STATUS_IN_PROCESS},
+		{"REQUEST_STATUS_NOT_FOUND", http.StatusNotFound, protocol.REQUEST_STATUS_NOT_FOUND},
+		{"REQUEST_STATUS_REJECTED", http.StatusBadRequest, protocol.REQUEST_STATUS_REJECTED},
+		{"REQUEST_STATUS_CONGESTION", http.StatusServiceUnavailable, protocol.REQUEST_STATUS_CONGESTION},
+	}
+	for i := range tests {
+		cTest := tests[i] // this is so that we can run tests in parallel, see https://gist.github.com/posener/92a55c4cd441fc5e5e85f27bca008721
+		t.Run(cTest.name, func(t *testing.T) {
+			t.Parallel()
+			require.Equal(t, cTest.expect, translateStatusToHttpCode(cTest.status), fmt.Sprintf("%s was translated to %d", cTest.name, cTest.expect))
+		})
+	}
+}
+
+func Test_HttpServer_writeMembuffResponse(t *testing.T) {
+	expectedResponse := (&client.SendTransactionResponseBuilder{
+		RequestStatus:      protocol.REQUEST_STATUS_COMPLETED,
+		TransactionReceipt: nil,
+		TransactionStatus:  protocol.TRANSACTION_STATUS_COMMITTED,
+		BlockHeight:        1,
+		BlockTimestamp:     primitives.TimestampNano(time.Now().Nanosecond()),
+	}).Build()
+
+	rec := httptest.NewRecorder()
+	writeMembuffResponse(rec, expectedResponse, http.StatusOK, "hello")
+
+	require.Equal(t, http.StatusOK, rec.Code, "code value is not equal")
+	require.Equal(t, "application/membuffers", rec.Header().Get("Content-Type"), "should have our content type")
+	require.Equal(t, "hello", rec.Header().Get("X-ORBS-CODE-NAME"), "should have correct x-orbs")
+	responseFromBody := client.SendTransactionResponseReader(rec.Body.Bytes())
+	test.RequireCmpEqual(t, expectedResponse, responseFromBody, "body response and pre-done response are not equal")
+}
+
+func Test_HttpServer_writeTextResponse(t *testing.T) {
+	rec := httptest.NewRecorder()
+	writeTextResponse(rec, "hello test", http.StatusAccepted)
+	require.Equal(t, http.StatusAccepted, rec.Code, "code value is not equal")
+	require.Equal(t, "plain/text", rec.Header().Get("Content-Type"), "should have our content type")
+	require.Equal(t, "hello test", rec.Body.String(), "should have text value")
+}
+
+func Test_HttpServer_SendTxHandler_Basic(t *testing.T) {
```

Comment:
please separate unit tests (testing the internal functions) from component tests (testing the handlers), put each type into own file